### PR TITLE
feat(auth-components): Add oxygen react auth components

### PR DESCRIPTION
**Pull request by movinsilva**
### Purpose

- Add SignIn compound component.
     - SignInTypography
     - SignInPaper
     - SignInTextField
     - SignInButton
     - SignInLink
     - SignInDivider
     - SignInLink

### Related Issues
- [☂️ MVP](https://github.com/asgardeo/web-ui-sdks/issues/6)

### Related PRs
- N/A

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: packages/react/src/oxygen-ui-react-auth-components/SignInButton/sign-in-button.scss [hunk 1]
```diff
@@ -0,0 +1,21 @@
+.oxygen-sign-in-button {
+  margin: 16px 0;
+}
+
+.oxygen-sign-in-button-social {
+  background: white !important;
+  color: black !important;
```
Comment (pavinduLakshan) [inline packages/react/src/oxygen-ui-react-auth-components/SignInButton/sign-in-button.scss:7]: Is marking these css properties as !important intentional? wouldn't end users be able to override the component styles?

Comment (movinsilva) [reply packages/react/src/oxygen-ui-react-auth-components/SignInButton/sign-in-button.scss]: It needs to be updated.
Addressed this with 39e975757996aea484c759adb1d33e4347b5588a.

File: packages/react/package.json [hunk 2]
```diff
@@ -44,12 +44,14 @@
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "sass": "^1.75.0",
-    "stylelint": "^16.4.0",
+    "stylelint": "15.1.0",
```
[the remark this answers is not in the document]
Comment (movinsilva) [reply packages/react/package.json:47]: Yes. 
@wso2/stylelint-config seems to not support the newest stylelint version.

File: packages/react/src/oxygen-ui-react-auth-components/SignInAlert/sign-in-alert.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.OxygenSignInAlert {
```
Comment (dasuni-30) [inline packages/react/src/oxygen-ui-react-auth-components/SignInAlert/sign-in-alert.scss:1]: Let's add the license headers

File: packages/react/src/oxygen-ui-react-auth-components/SignInAlert/SignInAlert.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+const SignInAlert: ForwardRefExoticComponent<SignInAlertProps> & WithWrapperProps = forwardRef(
+  <C extends ElementType>(props: SignInAlertProps<C>, ref: MutableRefObject<HTMLHRElement>): ReactElement => {
+    const {className, error, info, warning, color, icon, ...rest} = props;
+
+    const classes: string = clsx(`Oxygen${COMPONENT_NAME}`, className);
+
+    let extendedColor: string = color;
+    if (!color) {
+      if (error) {
+        extendedColor = 'error';
```
Comment (dasuni-30) [inline packages/react/src/oxygen-ui-react-auth-components/SignInAlert/SignInAlert.tsx:46]: Shall we add constants to these?

Comment (movinsilva) [reply packages/react/src/oxygen-ui-react-auth-components/SignInAlert/SignInAlert.tsx]: added with 1b94e9f12c495f46357fabc82822987a0c7f38a5

File: packages/react/src/oxygen-ui-react-auth-components/SignInAlert/sign-in-alert.scss (hunk 3 above)
Comment (movinsilva) [reply packages/react/src/oxygen-ui-react-auth-components/SignInAlert/sign-in-alert.scss:19]: added with 06f7ef89e7d1013b7029e5ee47bd6da2fd6e07e1

File: packages/react/src/oxygen-ui-react-auth-components/SignInButton/SignInButton.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+const SignInButton: ForwardRefExoticComponent<SignInButtonProps> & WithWrapperProps = forwardRef(
+  <C extends ElementType>(props: SignInButtonProps<C>, ref: MutableRefObject<HTMLButtonElement>): ReactElement => {
+    const {className, variant, social, ...rest} = props;
+
+    let classes: string = clsx(`Oxygen${COMPONENT_NAME}`, className);
+    if (social) {
+      classes = clsx(classes, `Oxygen${COMPONENT_NAME}Social`);
+    }
+
+    return <Button ref={ref} className={classes} fullWidth variant="contained" {...rest} />;
```
Comment (dasuni-30) [inline packages/react/src/oxygen-ui-react-auth-components/SignInButton/SignInButton.tsx:41]: Let's stick to one of single quotes or double quotes

File: packages/react/src/oxygen-ui-react-auth-components/SignInButton/sign-in-button.scss [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+.OxygenSignInButton {
+  margin: 16px 0;
+}
+
+.OxygenSignInButtonSocial {
+  background: white;
+  color: black;
+  margin: 0;
+  padding: 8px 0;
+  /* stylelint-disable-next-line color-function-notation */
```
Comment (dasuni-30) [inline packages/react/src/oxygen-ui-react-auth-components/SignInButton/sign-in-button.scss:28]: Shall we not suppress the stylelint rules?

**Review by dasuni-30 — DISMISSED** [review]

File: packages/react/src/oxygen-ui-react-auth-components/SignInButton/sign-in-button.scss (hunk 6 above)
Comment (movinsilva) [reply packages/react/src/oxygen-ui-react-auth-components/SignInButton/sign-in-button.scss]: removed with b2837c37c77e5a9c0a1bf501ec5e78d5102b5dd2

**Review by dasuni-30 — APPROVED** [review]

File: packages/react/src/oxygen-ui-react-auth-components/SignInButton/SignInButton.tsx (hunk 5 above)
Comment (movinsilva) [reply packages/react/src/oxygen-ui-react-auth-components/SignInButton/SignInButton.tsx:41]: eslint configs that we are using is defined to be this way. component props are using double quotes.

**Pull request merged**